app/assets/javascripts/views/edit.js

```diff
@@ -0,0 +1,19 @@
+var editDocument = document.querySelectorAll('.edit_document')
+if (editDocument) {
+  var beforeUnloadEvent = function (e) {
+    e.preventDefault()
+    e.returnValue = true
+  }
+
+  document.querySelectorAll('[data-warn-before-unload=true]').forEach(function (element) {
+    element.addEventListener('change', function () {
+      window.addEventListener('beforeunload', beforeUnloadEvent)
+    })
+  })
+
+  document.querySelectorAll('button[type=submit]').forEach(function (element) {
```

Comment:
I think rather than listening for a click event on this button we can add an eventListener on the form submit event